session.go

```diff
@@ -686,6 +686,7 @@ func (s *session) onDisconnect() {
 		s.messageOut = nil
 	}
 
+	<-s.messageIn
```

Comment:
will this not also block forever if there is no message in messageIn?

Reply:
You're right. The tests made me realized it. I tried doing a non-blocking receive instead, i.e.:

`// non-blocking receive to drain messageIn
select {
case <-s.messageIn:
// do nothing
default:
// do nothing
}
`
But this is not successful in solving the original problem (i.e. somehow, the select must be falling into the default cause without reading anything from messageIn). I'm trying to come up with another fix... Let me know if you have any ideas?

Reply:
@cbusbey See PR #403 for another possible fix. That one passes all required tests in Travis (but it's still running some of the "allowed failures" test).